cmd/pitr/recoverer/recoverer.go

```diff
@@ -359,6 +367,28 @@ func (r *Recoverer) setBinlogs() error {
 			return errors.Wrapf(err, "check if '%s' is a subset of '%s", r.startGTID, binlogGTIDSet)
 		}
 		if subResult != r.startGTID {
+			binlogName := binlog
+			if len(list)-1 < i+1 {
+				binlogName = list[i+1]
+			}
+			infoObj, err := r.storage.GetObject(binlogName + "-gtid-set")
+			if err != nil {
+				log.Println("Can't get binlog object with gtid set. Name:", binlog, "error", err)
+				continue
+			}
+			content, err := ioutil.ReadAll(infoObj)
+			if err != nil {
+				return errors.Wrapf(err, "read %s gtid-set object", binlog)
+			}
+			set, err := getExtendGTIDSet(string(content), r.startGTID)
```

Comment:
why we exclude all transactions since `r.startGTID` till the end of binlog?
maybe we need to exclude all transactions from the beginning of binlog till `r.startGTID`?